subprojects/reporting/src/main/java/org/gradle/api/reporting/ReportContainer.java

```diff
@@ -92,14 +89,17 @@ public ImmutableViolationException() {
     boolean isEmpty();
 
     @Incubating
-    @OutputDirectories
+    @Internal
+    @Deprecated
     Map<String, File> getEnabledDirectoryReportDestinations();
 
     @Incubating
-    @OutputFiles
+    @Internal
+    @Deprecated
     Map<String, File> getEnabledFileReportDestinations();
 
     @Incubating
-    @Input
+    @Internal
+    @Deprecated
     SortedSet<String> getEnabledReportNames();
```

Comment:
Didn't we say we'd remove these methods and add `getEnabledReports()` instead?

Reply:
Done.

Reply:
I was too fast. If I move `getEnabledReports()` here, then I get a property validation failure:
```
Execution failed for task ':diagnostics:validateTaskProperties'.
> Task property validation failed. See https://docs.gradle.org/4.7-20180222002812+0000/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Task type 'org.gradle.api.reporting.dependencies.HtmlDependencyReportTask': property 'reports.enabledReports.<key>.destination' is not annotated with an input or output annotation.
```

The static validation is right, since the static type does not have an annotation. At runtime though, everything will work fine.
Any ideas how to fix this? I am also fine with leaving `getEnabledReports()` in `DefaultReportContainer` and removing the three deprecated methods here.

Reply:
FTR, we discussed to use `@Internal` on `getDestination()` until we can have something like `@Abstract` to denote properties that need their types specified further down the inheritance chain.